lib/ios/app.nit

```diff
@@ -0,0 +1,187 @@
+
+		// TODO copy or pin the strings when needed
+		app_nit_ios_argv = malloc(argc * sizeof(char*));
+		app_nit_ios_argv[0] = program_name;
+		for (int i = 0; i < argc; i ++) {
+			String arg = Sequence_of_String__index(argv, i);
+			app_nit_ios_argv[i+1] = String_to_cstring(arg);
+		}
+	`}
+
+	# Register `self` globally in C so it can be retrieved from iOS callbacks
+	private fun register_globally in "ObjC" `{
+		App_incr_ref(recv);
+		app_nit_ios_app = recv;
+	`}
+
+	# Entry point to the iOS framework
+	private fun ui_application_main: Bool import did_finish_launching_with_options,
+	will_finish_launching_with_options,
+	will_resign_active, did_enter_background, will_enter_foreground,
+	did_become_active, will_terminate, ui_application=, app_delegate= in "ObjC" `{
+
+		@autoreleasepool {
+			return UIApplicationMain(app_nit_ios_argc, app_nit_ios_argv,
+				nil, NSStringFromClass([AppDelegate class]));
+		}
+	`}
+
+	# Very first custom code to be executed
+	fun will_finish_launching_with_options: Bool do return true
+
+	# Override point for customization after application launch.
+	#
+	# Right before the application is displayed to the user.
+	fun did_finish_launching_with_options: Bool do return true
+
+	# The application is about to move from active to inactive state
+	#
+	# This can occur for certain types of temporary interruptions
+	# (such as an incoming phone call or SMS message) or when the
+	# user quits the application and it begins the transition to
+	# the background state.
+	# Use this method to pause ongoing tasks, disable timers, and
+	# throttle down OpenGL ES frame rates. Games should use this
+	# method to pause the game.
+	fun will_resign_active do end
+
+	# The application just left foreground it can be suspended at any time
+	#
+	# Use this method to release shared resources, save user data,
+	# invalidate timers, and store application state to restore your
+	# application to its current state in case it is terminated later.
+	#
+	# If your application supports background execution, this method
+	# is called instead of `will_terminate` when the user quits.
+	fun did_enter_background do end
+
+	# The application will enter the foreground
+	#
+	# Called as part of the transition from the background to the
+	# inactive state; here you can undo many of the changes made
+	# on entering the background.
+	fun will_enter_foreground do end
+
+	# The application just became active
+	#
+	# Restart any tasks that were paused (or not yet started) while
+	# the application was inactive. If the application was previously
+	# in the background, optionally refresh the user interface.
+	fun did_become_active do end
+
+	# The application is about to terminate (not suspended)
+	#
+	# Save data if appropriate. See also applicationDidEnterBackground:.
+	fun will_terminate do end
+end
+
+app.register_args(program_name.to_cstring, args.length, args)
+app.register_globally
+
+var ret = app.ui_application_main
+exit if ret then 0 else 1
```

Comment:
When I read this one line, I cannot stop to thing that maybe science has gone too far...